### PR TITLE
[Refactor]Modify patchembed

### PR DESCRIPTION
**Pull request by HIT-cwh**
Hi! It's a common case to implement PatchEmbed and HybridEmbed by conv2d. This pr allows users to set hyper-parameter in conv2d, such as kernel_size, stride, padding, dilation and something like that, as their wish. What's more, as PatchEmbed and HybridEmbed may also be used in other backbones, I move them from /backbone/vision_transformer.py to /utils/embed.py.
Pls take a look.

### CODE REVIEW

File: mmcls/models/utils/embed.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+class PatchEmbed(BaseModule):
+    """Image to Patch Embedding V2.
+
+    We use a conv layer to implement PatchEmbed.
+    Args:
+        img_size (int | tuple): The size of input image. Default: 224
+        in_channels (int): The num of input channels. Default: 3
+        embed_dims (int): The dimensions of embedding. Default: 768
+        norm_cfg (dict, optional): Config dict for normalization layer.
```
Comment (ycxioooong) [inline mmcls/models/utils/embed.py:17]: Default: None

File: mmcls/models/utils/embed.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        self.num_patches = h_out * w_out
+
+        if norm_cfg is not None:
+            self.norm = build_norm_layer(norm_cfg, embed_dims)[1]
+        else:
+            self.norm = None
+
+    def forward(self, x):
+        B, C, H, W = x.shape
+        # FIXME look at relaxing size constraints
```
Comment (ycxioooong) [inline mmcls/models/utils/embed.py:68]: Please check the FIXME again

File: mmcls/models/utils/embed.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                img_size = to_2tuple(img_size[0])
+            assert len(img_size) == 2, \
+                f'The size of image should have length 1 or 2, ' \
+                f'but got {len(img_size)}'
+
+        self.img_size = img_size
+        self.backbone = backbone
+        if feature_size is None:
+            with torch.no_grad():
+                # FIXME this is hacky, but most reliable way of
```
Comment (ycxioooong) [inline mmcls/models/utils/embed.py:123]: Please check the FIXME again

Comment (HIT-cwh) [reply mmcls/models/utils/embed.py:123]: I think it's hacky, but the most reliable way of determining the exact dim of the output feature map for all networks. If someone can find a better way, pls feel free to fix me.

**Review by ycxioooong — APPROVED** [review]
LGTM now
